From 48ee2899262982c0f450243e3482e446321604e0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Fri, 21 Apr 2017 21:57:43 +0100
Subject: [PATCH app/xinit] Provide stateless approach to startx for GNOME/XFCE

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 xinitrc.cpp | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/xinitrc.cpp b/xinitrc.cpp
index 14d3cbc..2887769 100644
--- a/xinitrc.cpp
+++ b/xinitrc.cpp
@@ -48,8 +48,18 @@ if [ -d XINITDIR/xinitrc.d ] ; then
 	unset f
 fi
 
-TWM &
-XCLOCK -geometry 50x50-1+1 &
-XTERM -geometry 80x50+494+51 &
-XTERM -geometry 80x20+494-0 &
-exec XTERM -geometry 80x66+0+0 -name login
+if [ -x /usr/bin/gnome-shell ]; then
+    export XDG_CURRENT_DESKTOP="GNOME-Flashback"
+    exec gnome-session  --session=gnome-flashback-metacity
+fi
+
+
+if [ -x /usr/bin/startxfce4 ] ; then
+    exec startxfce4
+fi
+
+
+echo "No usable desktops found. Please customise ~/.xinitrc"
+echo -e "Alternatively, install the GNOME experience:\n"
+echo "    swupd bundle-add desktop"
+echo "    swupd update"
-- 
2.34.0

