From 5f17ad981110da61ceaff65a8f867d9802b10c77 Mon Sep 17 00:00:00 2001
From: Francisco Boni <boboniboni@gmail.com>
Date: Mon, 6 Dec 2021 10:13:53 -0300
Subject: [PATCH] Remove unset DBUS_SESSION_BUS_ADDRESS

---
 startx.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/startx.cpp b/startx.cpp
index 31187b9..4977f35 100644
--- a/startx.cpp
+++ b/startx.cpp
@@ -11,7 +11,6 @@ XCOMM
 XCOMM Site administrators are STRONGLY urged to write nicer versions.
 XCOMM
 
-unset DBUS_SESSION_BUS_ADDRESS
 unset SESSION_MANAGER
 
 #ifdef __APPLE__
-- 
2.34.0

